cmd/cb-event-forwarder/main.go

```diff
@@ -387,14 +389,11 @@ func messageProcessingLoop(uri, queueName, consumerTag string) error {
 		select {
 		case outputError := <-outputErrors:
 			log.Errorf("ERROR during output: %s", outputError.Error())
-
-			// hack to exit if the error happens while we are writing to a file
-			if config.OutputType == FileOutputType || config.OutputType == SplunkOutputType || config.OutputType == HTTPOutputType || config.OutputType == S3OutputType || config.OutputType == SyslogOutputType {
-				log.Error("File output error; exiting immediately.")
-				c.Shutdown()
-				wg.Wait()
-				os.Exit(1)
-			}
+		case signal := <-outputSignals:
+			log.Errorf("%s- exiting immediately.", signal)
+			c.Shutdown()
+			wg.Wait()
+			os.Exit(1)
 
 		case closeError := <-c.connectionErrors:
```

Comment:
Some customers are reporting errors where we're getting connectionErrors and then exiting as well.  We should probably not exit here?

Reply:
We know it's this call and not the error call based on the logs.

Reply:
Case's are exclusive in golang so this should avoid exiting during runtime errors and only exit when there is an signal to the daemon to shutdown like SIGTERM